src/format/opamFile.ml

```diff
@@ -823,10 +826,13 @@ module ConfigSyntax = struct
                (kind,criterion)::List.remove_assoc kind t.solver_criteria }
   let with_solver t solver = { t with solver = Some solver }
 
+  let backup_url_ocamlpro = "http://opam.ocamlpro.com/backup"
```

Comment:
@dbuenzli @lefessan, please keep the conversation technical and focussed to the PR at hand please.

My comments:

- It is very useful to have a backup URL feature in OPAM, but I agree with @AltGr that this needs to either be per-remote and configurable, just as other remote-specific URLs are.  A global setting doesn't work well with the design of remotes, which can be in local infrastructure.
- Regarding opam.ocaml.org vs opam.ocamlpro.com, we must at a minimum remain consistent.  The current behaviour of `opam.ocaml.org` is that the archive service can be set to a local service very easily at init time, and nothing is baked into the tool beyond the initial default.  Baking in another domain that is non-ocaml.org just makes this much more difficult.  If possible, it would be very helpful to have instructions on how to set up the backup service, and then anyone can run this service (including --and we would be most grateful to -- OCamlPro).

In general, it is helpful to think of ocaml.org as just one of many "OPAM distributions" that combines OPAM and the ocaml/opam-repository and associated tools like opam2web into one service.  There are other emerging consumers in the open source world (such as the Coq community), industrial users whose codebases arent public, and I would hope commercially supported distributions who can all build services using the same tools. The design of PRs coming into OPAM should reflect that configurability and customisability and not bake in centralised choices where possible.